### ISSUE TITLE
aedes-5 RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by cordovapolymer** [mcollina/aedes#77]
I'm trying to get aedes working with QOS 0 and then move to QOS 2. I'm using the example.js with {concurrency: 1000} option to run aedes server.

I've encountered this issue, which happens on all QOS levels:

Aedes normally consumes 50-60MB of RAM, then one of websocket clients(`client2` here) gets disconnected, and when it reconnects aedes starts eating memory till it runs into `aedes-5 RangeError: Maximum call stack size exceeded`

```
aedes-5 client error client2 keep alive timeout Error: keep alive timeout
aedes-5     at keepaliveTimeout (/home/user/app/node_modules/aedes/lib/handlers/connect.js:44:28)
aedes-5     at Timeout.timerWrapper [as _onTimeout] (/home/user/app/node_modules/retimer/retimer.js:21:16)
aedes-5     at ontimeout (timers.js:365:14)
aedes-5     at tryOnTimeout (timers.js:237:5)
aedes-5     at Timer.listOnTimeout (timers.js:207:5)
aedes-5 message from client client1
aedes-5 message from client client1
aedes-5 message from client client1
aedes-5 message from client client1
aedes-5 new client client2
```

```
aedes-5 RangeError: Maximum call stack size exceeded
aedes-5     at SubscribeState.storeSubscriptions (/home/user/app/node_modules/aedes/lib/handlers/subscribe.js:57:29)
aedes-5     at work (/home/user/app/node_modules/fastfall/fall.js:105:23)
aedes-5     at Aedes.defaultAuthorizeSubscribe [as authorizeSubscribe] (/home/user/app/node_modules/aedes/aedes.js:298:3)
aedes-5     at SubscribeState.authorize (/home/user/app/node_modules/aedes/lib/handlers/subscribe.js:39:17)
aedes-5     at Holder.work (/home/user/app/node_modules/fastfall/fall.js:105:23)
aedes-5     at SubscribeState.compiled (/home/user/app/node_modules/fastfall/fall.js:55:17)
aedes-5     at SubscribeState.doSubscribe (/home/user/app/node_modules/aedes/lib/handlers/subscribe.js:34:16)
aedes-5     at makeCall (/home/user/app/node_modules/fastseries/series.js:113:10)
aedes-5     at SubscribeState.ResultsHolder.release (/home/user/app/node_modules/fastseries/series.js:94:9)
aedes-5     at work (/home/user/app/node_modules/fastfall/fall.js:121:25)
```
, after which aedes stops receiving or sending messages

I tried to increase stack size with `--stack-size` node argument, it suppressed the error but didn't solve it, aedes just stops receiving and sending messages after some period, although it reports that clients are connecting.

It seems that the issue has been described in https://github.com/mcollina/aedes/issues/66 and https://github.com/mcollina/aedes/issues/60 
In my case I'm not running it under a heavy load yet, just several client are connected and it occurs after a while on message publish, I use latest node 6.9.1 and latest MQTT.js 2.0.1 on the client side.

A way to reproduce the error- connect several websocket clients which constantly publish messages, the turn off the connection for one of them wait for several seconds and turn it back on, aedes will die on reconnection.

**update: it only occurs when `clean:false` option is set. When its' unset server stops receiving or sending messages without any error, it just shows that clients are connected but doesn't receive or send message without restart**

**Comment by cordovapolymer** [comment]
@mcollina any hope to solve this issue?

**Comment by mcollina** [comment]
Just to clarify: this is open source. I accept contributions, and I review them swiftly.
So, when and if I will have time, I will adress this issue if no one had.
Complaining about it does not solve the bug.
I am not paid to maintain this project, and I do it in my own time.

**Comment by cordovapolymer** [comment]
> Just to clarify: this is open source. I accept contributions, and I review them swiftly.
> So, when and if I will have time, I will adress this issue if no one had.
> Complaining about it does not solve the bug.
> I am not paid to maintain this project, and I do it in my own time.

LOL, we appreciate it, when I was considering what to use **mosca** or **aedes** I followed your recommendation in the mosca issue list to use aedes because its design is better and its implementing considering mosca experience, Although I've now switched to mosca I like aedes design decisions and it's support of QOS2 so I'd like to use it in the future.

If you could give me a tip how to debug this issue, we'd try to help solving it.

**Comment by mcollina** [comment]
@cordovapolymer it would be really awesome to have a script to reproduce this. Would you mind uploading one?

**Comment by cordovapolymer** [comment]
the issue has been resolved, running 0.27 and newer version for more than a week in production
